script/verdaccio.js

```diff
@@ -0,0 +1,220 @@
+    spawnOrFail,
+    process,
+    shouldContinuePrompt
+  } = require('./utilities');
+  const path = require('path');
+  const cp = require('child_process');
+  const fs = require('fs');
+  const PORT = 4873;
+  const verdaccioEndpoint = `http://localhost:${PORT}`;
+  const verdaccioAppDir = path.join(__dirname, '../../verdaccio_app');
+  const sdkTestDir = './test-chime-sdk-js';
+  const SDK_APP_NAME = 'amazon-chime-sdk-js';
+  const applicationProcesses = [];
+
+  const cleanupBeforeExit = () => {
+    logger.log('Terminating processes...');
+    applicationProcesses.map(process => {
+      logger.log(`Killing process: ${process.pid}`);
+      spawnOrFail('kill', [`kill -9`, `${process.pid}`]);
+    });
+    spawnOrFail('ps', [
+      `-ef | grep -E '(webpack|verdaccio)' | grep -v grep | awk '{ print $2 }' | xargs kill -9`
+  ]);
+    process.exit(0);
+  };
+  
+  process.on('SIGTERM', () => {
+    logger.warn("Received SIGTERM, cleaning up.")
+    cleanupBeforeExit();
+  });  
+  process.on('SIGINT', () => {
+    logger.warn("Received SIGINT, cleaning up.")
+    cleanupBeforeExit();
+  });
+  process.on('error', (e) => {
+      logger.error(`Error received: ${e}`);
+      cleanupBeforeExit();
+  })
+  
+  // Create verdaccio testing directory and cd in to it
+  if (!fs.existsSync(verdaccioAppDir)) {
+    logger.log(`Creating verdaccio directory: ${verdaccioAppDir}`);
+    fs.mkdirSync(verdaccioAppDir);
+  }
+  
+  //  Go to verdaccio dir
+  logger.log(`Change directory to verdaccio directory: ${verdaccioAppDir}`);
+  process.chdir(verdaccioAppDir);
+
+  // Initiate basic node package
+  logger.log(`Creating basic node package`);
+  spawnOrFail('npm', ['init', '-f', '-y']);
+  
+  // Install verdaccio within new package
+  logger.log(`Installing Verdaccio`);
+  spawnOrFail('npm', ['i', 'verdaccio']);
+  
+  generateVerdaccioConfigFile();
+  
+  // Start verdaccio application
+  try {
+    startVerdaccioApp();
+  } catch (e) {
+    logger.error('Failed to start verdaccio', e);
+  }
+  
+  // Make SDK test directory
+  if (!fs.existsSync(sdkTestDir)) {
+    logger.log(`Creating SDK testing directory: ${sdkTestDir}`);
+    fs.mkdirSync(sdkTestDir);
+  } else {
+    logger.log(`SDK testing directory already exists: ${sdkTestDir}`);
+  }
+
+  logger.log(`Navigate to SDK testing directory: ${sdkTestDir}`);
+  process.chdir(sdkTestDir);
+  
+  // Clone SDK from github
+  logger.log(`Cloning ${SDK_APP_NAME} into ${sdkTestDir}`);
+  
+  //Clean up
+  spawnOrFail('rm', ['-rf', `${SDK_APP_NAME}`]);
+  spawnOrFail('git', [
+    'clone',
+    'git@github.com:aws/amazon-chime-sdk-js.git'
+  ]);
+  // navigate in to cloned SDK
+  process.chdir(path.join(`./${SDK_APP_NAME}`));
+  
+  // Update package.json version
+  let package = fs.readFileSync('package.json', 'utf-8');
+
+  const packageJson = JSON.parse(package);
+  const originalVersion = packageJson.version;
+  const versions = originalVersion.split('.').map(x => parseInt(x));
+  versions[2] += 1;
```

Comment:
Do we want to prompt for the MAJOR, MINOR or PATCH in the version here, or for testing its fine to just increase the patch?

Reply:
We may not need to prompt for MAJOR MINOR or PATCH and instead just provide a prompt to enter the version to be used.

Reply:
We do this for the components release, and IMO its easier to have patch, minor, major. Less room for error

Reply:
I think no update is made here right to get the explicit major, minor and patch or version directly as @ltrung said?

Reply:
This is not correct though because MCM can do MAJOR, MINOR, PATCH or manual version.